File: DictionaryTest.c
//-----------------------------------------------------------------------------
//Jonathan Molina
//jomolina
//CS12B
//DictionaryTest.c
//Test  for the Dictionary ADT
//-----------------------------------------------------------------------------
//import all the stuff we might use or need
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"


//tests the insert function
int main(int argc, char* argv[]){
   Dictionary A = newDictionary();
   char* k;
   char* v;
 
/*   char* word1[] = {"hey","perro","twin","california","million","billion","seven"};
   char* word2[] = {"3","5","blah7","7","7","9","O"};
   int i;
//tests the insert function
   for(i=0; i<7; i++){
      insert(A, word1[i], word2[i]);
*///Insert does not work insert does not get called
  // }
 // insert(A, "five", "glow"); // error: duplicate key the error works well exception gets thrown
  // printDictionary(stdout, A);


 delete(A, "one");
   delete(A, "three");
   delete(A, "seven");

   printDictionary(stdout, A);

   for(i=0; i<7; i++){
      k = word1[i];
      v = lookup(A, k);
      printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   }

   // delete(A, "one");  // error: key not found error gets thrown functionality works well
   //test for the empty
    /* printf("%s\n", (isEmpty(A)?"true":"false"));
     printf("%d\n", size(A));
     makeEmpty(A);
     printf("%s\n", (isEmpty(A)?"true":"false"));
    */ //empty does not work error when inserting or deleting or in makeempty
//second test for insert
 char* word1[] = {"one","two","three","four","five","six","seven"};
   char* word2[] = {"foo","bar","blah","galumph","happy","sad","blue"};
   int i;

   for(i=0; i<7; i++){
      insert(A, word1[i], word2[i]);
   }

   printDictionary(stdout, A);


//second test for make empty     
     printf("%s\n", (isEmpty(A)?"true":"false"));
     printf("%d\n", size(A));
     makeEmpty(A);
     printf("%s\n", (isEmpty(A)?"true":"false"));
     freeDictionary(&A);
  
     return(EXIT_SUCCESS)



}

File: DictionaryTest.java
import java.util.*;//imports utilities we might use
/*===========================================*/
/*Jonathan Molina                            */    
/*PA 3 CS12B                                 */
/*jomolina                                   */
/* This Assignment checks all functions work */
/*===========================================*/   
public class DictionaryTest{

   public static void main(String[] args){
      
      String v;
      Dictionary A = new Dictionary();
     // A.insert("1","word");
     // A.insert("2","face");
     // A.insert("3","dog");
     // A.insert("4","hair");
     // A.insert("5","hi");
     // A.insert("6","hello");
     // A.insert("7","why");
      
     // v = A.lookup("1");
     //Prints keys found and not found correctly. 
     // System.out.println("Key = 1" +( v == null? "not found" :(" Value = " +v)));
     // v = A.lookup("3");
     // System.out.println("Key = 3" +( v == null? "not found" :(" Value = " +v)));
     // v = A.lookup("7");
     // System.out.println("Key = 7" +( v == null? "not found" :(" Value = " +v)));
     //  v = A.lookup("8");
     // System.out.println("key=8 "+(v==null?"not found":("value="+v)));
     // System.out.println();      
     System.out.println(A.size());//size works but check the if output is correct     
     //System.out.println(A.isEmpty());//works correctly it prints false when it is empty
    // System.out.println(A);//prints the values 7, 6, 5, 4, 3, 2, 1; 
    // A.makeEmpty();//empties correctly
    // System.out.println(A.isEmpty());//correctly checks if is empty after the function makeEmpty is called
   
    System.out.println(A);// keys get printed backwards
  

     
//properly deletes stuff from the Node list
//A.delete("1");
//System.out.println(A.size());
//A.delete("3");
//System.out.println(A.size());
//A.delete("7");
//System.out.println(A.size());
//System.out.println(A);
//System.out.println(A.size());
//checks error exceptions
//A.insert("2","f");
//A.delete("8"); they work out properly
//
//checks found and not found with different keys values
//v = A.lookup("7");
//System.out.println("Key = 7" +( v == null? "not found" :(" Value = " +v)));
//v = A.lookup("3");
//System.out.println("Key = 3" +( v == null? "not found" :(" Value = " +v)));
//v = A.lookup("5");
//System.out.println("Key = 5" +( v == null? "not found" :(" Value = " +v)));
//System.out.println(A);


//checks if size is being subtracted correctly after deleting keys and checks if deletes still works correctly
//A.delete("1");
//System.out.println(A.size());
//A.delete("3");
//System.out.println(A.size());
//A.delete("7");
//System.out.println(A.size());
System.out.println(A);



//checks if insert is working properly
      A.insert("1","a");
      A.insert("2","b");
      A.insert("3","c");
      A.insert("4","d");
      A.insert("5","e");
      A.insert("6","f");
      A.insert("7","g");
      System.out.println(A);// prints in correct order 
       }
}
